pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -505,6 +525,58 @@ func (r *ReconcileIBMBlockCSI) reconcileServiceAccount(instance *ibmblockcsi.IBM
 	return nil
 }
 
+func (r *ReconcileIBMBlockCSI) getRestartedAtAnnotation(Annotations map[string]string) (string, string){
+	restartedAt := fmt.Sprintf("%s/restartedAt", oconfig.APIGroup)
+	for key, element := range Annotations {
+		if key == restartedAt {
+			return key, element
+		}
+	}
+	return "", ""
+}
+
+func (r *ReconcileIBMBlockCSI) getControllerK8sObject(instance *ibmblockcsi.IBMBlockCSI) (error, *appsv1.StatefulSet) {
+			controllerStatefulset := &appsv1.StatefulSet{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{
+				Name:      oconfig.GetNameForResource(oconfig.CSIController, instance.Name),
+				Namespace: instance.Namespace,
+			}, controllerStatefulset)
+		
+			if err != nil {
+				return err, controllerStatefulset
+			}
+			return err, controllerStatefulset
```

Comment:
Why do we need the if statements?

Reply:
Also, usually, when we return the error and another value, the error is the second value.